src/test/java/web/tests/FlipkartTest.java

```diff
@@ -168,19 +170,45 @@ public void testAddToCartFunctionality(String product) {
     }
 
     /**
-     *
-     * @param item :accepting shoes from json
+     * @param item     :accepting shoes from json
      * @param position :accepting position i.e to select 2nd position shoes
      */
     @ParameterizedTest
     @CsvFileSource(resources = "/testProductPageDesign.csv")
-    public void testProductPageDesign(String item,int position){
+    public void testProductPageDesign(String item, int position) {
         ProductPage productPage = new HomePage().popUpCancel().setShoes(item).searchShoes().OpenProductPage(position);
         productPage.scrollToBottom();
-        ArrayList<String> headers  = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
-        for(String headersInSelectionPage:headers){
+        ArrayList<String> headers = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
+        for (String headersInSelectionPage : headers) {
             softAssert.assertThat(productPage.isSelectionDisplayed(headersInSelectionPage)).
                     as("Json selected Headers is Not matching with selection page Headers").isTrue();
         }
     }
+
+    @Test
+    public void flipkartSocialMedia(){
+        HomePage homePage = new HomePage().popUpCancel();
+        SocialMediaPage socialMediaPage=null;
+        ArrayList<String> socialMediaLinks = new ArrayList<>(Arrays.asList(map.get("links")
+                .split(",")));
+        for (int i = 0; i < socialMediaLinks.size(); i++) {
+            System.out.println("*****links*****"+socialMediaLinks.get(i));
+            socialMediaPage =homePage.clickLink(socialMediaLinks.get(i));
+            if(socialMediaLinks.get(i).equals("Facebook")) {
+                socialMediaPage.checkFacebookElementVisible();
+            }
+            if(socialMediaLinks.get(i).equals("Twitter")) {
+                socialMediaPage.checkTwitterElementVisible();
+            }
+            if(socialMediaLinks.get(i).equals("YouTube")) {
+                socialMediaPage.checkYoutubeElementVisible();
+            }
+            String url = socialMediaPage.getSocialMediaUrl();
+                logger.info("********" + url);
+                logger.info("********" + socialMediaLinks.get(i).toLowerCase());
+                softAssert.assertThat(url.contains(socialMediaLinks.get(i).toLowerCase())).
+                        as("links of social media are not matching with the contents in url").isTrue();
```

Comment:
Lets use some other way to assert which is more optimised like :
softAssert.assertThat().matches()
softAssert.assertThat().contains
softAssert.assertThat().containPattern